src/graph/graph.c

```diff
@@ -420,6 +437,9 @@ int Graph_LabelTypeCount(const Graph *g) {
 void Graph_AllocateNodes(Graph *g, size_t n) {
 	assert(g);
 	DataBlock_Accommodate(g->nodes, n);
+	// Updated the required matrix size if this creation will trigger a resize.
+	size_t new_count = Graph_UncompactedNodeCount(g) + n;
```

Comment:
Because we're reusing deleted nodes, I believe `new_count` should be:

```
new_count = Graph_NodeCount() + Graph_DeletedNode();
if( (n - Graph_DeletedNode()) > 0) new_count += n - Graph_DeletedNode();
```